File: TrendAnalyzerServer/src/main/java/vorotilenko/trendanalyzerserver/clients/TradeInfo.java
package vorotilenko.trendanalyzerserver.clients;

/**
 * Stores data about trades
 */
public class TradeInfo {
    /**
     * Symbol
     */
    public String symbol;
    /**
     * Trade time in milliseconds
     */
    public long tradeTimeMillis;
    /**
     * Price of the trade
     */
    public double price;
    /**
     * Exchange name
     */
    public String exchange;

    /**
     * @param symbol Symbol
     * @param tradeTimeMillis Trade time in milliseconds
     * @param price Price of the trade
     * @param exchange Exchange name
     * */
    public TradeInfo(String symbol, long tradeTimeMillis, double price, String exchange) {
        this.symbol = symbol;
        this.tradeTimeMillis = tradeTimeMillis;
        this.price = price;
        this.exchange = exchange;
    }
}

File: TrendAnalyzer/app/src/main/java/vorotilenko/trendanalyzer/TradeInfo.kt
package vorotilenko.trendanalyzer

/**
 * Stores data about trades including time, price, symbol and exchange
 */
data class TradeInfo(
    /**
     * Symbol
     */
    val symbol: String,
    /**
     * Trade time in milliseconds
     */
    val tradeTimeMillis: Long,
    /**
     * Price of the trade
     */
    val price: Double,
    /**
     * Exchange name
     */
    val exchange: String
)
